skore/src/skore/item/cross_validation_item.py

```diff
@@ -299,15 +181,61 @@ def __init__(
         self.plot_bytes = plot_bytes
 
     @classmethod
-    def factory(
+    def factory(cls, *args, **kwargs):
+        """
+        Create a new CrossValidationItem instance.
+
+        Redirects to one of the underlying `factor_*` class methods depending
+        on the arguments
+
+        Returns
+        -------
+        CrossValidationItem
+            A new CrossValidationItem instance.
+        """
+        if args:
+            with contextlib.suppress(ItemTypeError):
+                return cls.factory_cross_validation_reporter(args[0])
+
+        return cls.factory_raw(*args, **kwargs)
+
+    @classmethod
+    def factory_cross_validation_reporter(cls, reporter: CrossValidationReporter):
+        """
+        Create a new CrossValidationItem instance from a CrossValidationReporter.
+
+        Parameters
+        ----------
+        reporter : CrossValidationReporter
+
+        Returns
+        -------
+        CrossValidationItem
+            A new CrossValidationItem instance.
+        """
+        if reporter.__class__.__name__ != "CrossValidationReporter":
```

Comment:
Why not:

```suggestion
        if not isinstance(reporter, CrossValidationReporter):
```

I don't see circular imports.

Reply:
I do get them:
```
ImportError while loading conftest '/home/auguste/Desktop/skore/skore/tests/conftest.py'.
tests/conftest.py:4: in <module>
    from skore.item.item_repository import ItemRepository
src/skore/__init__.py:7: in <module>
    from skore.project import Project, create, load
src/skore/project/__init__.py:3: in <module>
    from .create import create
src/skore/project/create.py:14: in <module>
    from skore.project.load import load
src/skore/project/load.py:6: in <module>
    from skore.item import ItemRepository
src/skore/item/__init__.py:9: in <module>
    from skore.item.cross_validation_item import CrossValidationItem
src/skore/item/cross_validation_item.py:18: in <module>
    from skore.sklearn.cross_validation import CrossValidationReporter
src/skore/sklearn/__init__.py:4: in <module>
    from skore.sklearn.train_test_split.train_test_split import train_test_split
src/skore/sklearn/train_test_split/train_test_split.py:11: in <module>
    from skore.project import Project
E   ImportError: cannot import name 'Project' from partially initialized module 'skore.project' (most likely due to a circular import) (/home/auguste/Desktop/skore/skore/src/skore/project/__init__.py). Did you mean: 'project'?
```

but it might be because you missed [this line](https://github.com/probabl-ai/skore/pull/847/files#diff-939ea0f3a7a59b88269e22268fddaecedba8d9f3212b0bc66f49e1d3bc4231caR22)?